src/Core/TileMesh.js

```diff
@@ -11,10 +11,12 @@ import RendererConstant from '../Renderer/RendererConstant';
 import OGCWebServiceHelper, { SIZE_TEXTURE_TILE } from '../Provider/OGCWebServiceHelper';
 import { is4326 } from './Geographic/Coordinates';
 
-function TileMesh(geometry, params) {
+function TileMesh(layer, geometry, params) {
     // Constructor
     THREE.Mesh.call(this);
 
+    this.layer = layer;
+
```

Comment:
What is the semantics of this layer ? which one is it ? I would rather keep it outside of TileMesh as with other meshes in itowns.

Reply:
It's the GeometryLayer, and as stated above it was already stored in TileMesh (and I think it's useful).

Reply:
For instance (here)[https://github.com/iTowns/itowns/blob/master/src/Core/MainLoop.js#L101].

And I'll repeat myself: this isn't a new thing. I simply moved the `layer` assignement inside the constructor.